traffic_ops/testing/api/v2/deliveryservices_test.go

```diff
@@ -0,0 +1,602 @@
+		t.Errorf("cannot PUT deliveryservice: %s", err.Error())
+	}
+	putDSResp := tc.UpdateDeliveryServiceNullableResponse{}
+	if err = makeV11Request(http.MethodPut, "deliveryservices/"+strconv.Itoa(*respID), bytes.NewBuffer(dsV11Body), &putDSResp); err != nil {
+		t.Errorf("cannot PUT 1.1 deliveryservice, failed to make request: %s", err.Error())
+	}
+	if !dsV13FieldsAreNil(putDSResp.Response[0]) || !dsV14FieldsAreNil(putDSResp.Response[0]) {
+		t.Errorf("PUT 1.1 expected 1.3 and 1.4 values to be nil, actual: non-nil %+v", putDSResp.Response[0])
+	}
+	getDS, _, err = TOSession.GetDeliveryServiceNullable(strconv.Itoa(*respID))
+	if err != nil {
+		t.Errorf("cannot GET deliveryservice: %s", err.Error())
+	}
+	if getDS.FQPacingRate == nil {
+		t.Errorf("expected FQPacingRate: %d, actual: nil", testDS.FQPacingRate)
+	} else if *getDS.FQPacingRate != testDS.FQPacingRate {
+		t.Errorf("expected FQPacingRate: %d, actual: %d", testDS.FQPacingRate, *getDS.FQPacingRate)
+	}
+	if getDS.MaxOriginConnections == nil {
+		t.Errorf("expected MaxOriginConnections: %d, actual: nil", testDS.MaxOriginConnections)
+	} else if *getDS.MaxOriginConnections != testDS.MaxOriginConnections {
+		t.Errorf("expected MaxOriginConnections: %d, actual: %d", testDS.MaxOriginConnections, *getDS.MaxOriginConnections)
+	}
+
+	// PUT 1.3 w/ 1.1 data, verify that 1.4 fields were preserved
+	putDSResp = tc.UpdateDeliveryServiceNullableResponse{}
+	if err = makeV13Request(http.MethodPut, "deliveryservices/"+strconv.Itoa(*respID), bytes.NewBuffer(dsV11Body), &putDSResp); err != nil {
+		t.Errorf("cannot PUT 1.3 deliveryservice, failed to make request: %s", err.Error())
+	}
+	if !dsV14FieldsAreNil(putDSResp.Response[0]) {
+		t.Errorf("PUT 1.3 expected 1.4 values to be nil, actual: non-nil %+v", putDSResp.Response[0])
+	}
+	getDS, _, err = TOSession.GetDeliveryServiceNullable(strconv.Itoa(*respID))
+	if err != nil {
+		t.Errorf("cannot GET deliveryservice: %s", err.Error())
+	}
+	if getDS.MaxOriginConnections == nil {
+		t.Errorf("expected MaxOriginConnections: %d, actual: nil", testDS.MaxOriginConnections)
+	} else if *getDS.MaxOriginConnections != testDS.MaxOriginConnections {
+		t.Errorf("expected MaxOriginConnections: %d, actual: %d", testDS.MaxOriginConnections, *getDS.MaxOriginConnections)
+	}
+
+	// DELETE+POST 1.1 again, so that 1.3 and 1.4 fields are back to nil/default
+	if _, err = TOSession.DeleteDeliveryService(strconv.Itoa(*respID)); err != nil {
+		t.Errorf("cannot DELETE deliveryservice: %s", err.Error())
+	}
+	postDSResp = tc.CreateDeliveryServiceNullableResponse{}
+	if err = makeV11Request(http.MethodPost, "deliveryservices", bytes.NewBuffer(dsV11Body), &postDSResp); err != nil {
+		t.Errorf("cannot POST 1.1 deliveryservice, failed to make request: %s", err.Error())
+	}
+	respID = postDSResp.Response[0].ID
+
+	// PUT 1.1 w/ 1.4 data - make sure 1.3 and 1.4 fields were ignored
+	putDSResp = tc.UpdateDeliveryServiceNullableResponse{}
```

Comment:
This whole bit about updating a Delivery Service created at API version 1.1 with data created for the v1.4 API submitted to the v1.1 API then checking that the resulting object properly ignored newer fields shouldn't be included, because this can't won't test the v1.1/1.4 API